### PR TITLE
rewrite some input rumble functions

### PR DESCRIPTION
**Pull request by punpckhdq**
more than 1 controller is untested but it will try its best to set the rumble state of extra controllers

### CODE REVIEW

**Review by Berthalamew — APPROVED** [review]

**Comment by Berthalamew** [comment]
Tested with kb/m and 2 controllers.
Vibration works correctly for controllers

File: xlive/Blam/Engine/game/player_vibration.cpp [hunk 1]
```diff
@@ -33,133 +33,127 @@ void __cdecl rumble_player_continuous(int32 user_index, real32 rumble_intensity_
...
-                        input_set_gamepad_rumbler_state(player->controller_index, state.wLeftMotorSpeed, state.wRightMotorSpeed);
-
-                        ASSERT(VALID_INDEX(player->controller_index, k_number_of_controllers));
-                        controller_rumble_enabled[player->controller_index] = true;
-                    }
-                }
-            }
-        }
-
-        for (uint32 i = 0; i < k_number_of_users; i++)
```
Comment (Berthalamew) [inline xlive/Blam/Engine/game/player_vibration.cpp:90]: why was this removed?

File: xlive/Blam/Engine/input/input_xinput.cpp [hunk 2]
```diff
@@ -73,33 +72,44 @@ void input_xinput_clear_rumble_state(void)
...
+		|| game_time_get_paused())
 	{
-		suppress_rumble = true;
+		global_suppress_rumble = true;
 	}
 
-	if (controller_button_state_get(_controller_index_0)->plugged_in)
+	for (uint32 device_index = 0; device_index < k_number_of_controllers; ++device_index)
 	{
-		g_xinput_vibration.wLeftMotorSpeed = (suppress_rumble ? 0 : g_vibration_state[_controller_index_0].wLeftMotorSpeed);
```
Comment (Berthalamew) [inline xlive/Blam/Engine/input/input_xinput.cpp:89]: rumble needs to be set to 0 if it's suppressed incase rumbling is already in progress.

File: xlive/Blam/Engine/game/player_vibration.cpp (hunk 1 above)
Comment (punpckhdq) [reply xlive/Blam/Engine/game/player_vibration.cpp:90]: restored in latest commit

File: xlive/Blam/Engine/input/input_xinput.cpp (hunk 2 above)
Comment (punpckhdq) [reply xlive/Blam/Engine/input/input_xinput.cpp:89]: handled in function

**Comment by Berthalamew** [comment]
Tested with kb/m and 3 controllers.
Vibration works correctly.
